at_client/lib/src/service/notification_service_impl.dart

```diff
@@ -408,4 +412,37 @@ class NotificationServiceImpl
   getStreamListenersCount() {
     return _streamListeners.length;
   }
+
+  @override
+  Future<AtNotification> fetch(String notificationId) async {
+    var notifyFetchVerbBuilder = NotifyFetchVerbBuilder()
+      ..notificationId = notificationId;
+    var atNotificationStr = await _atClient
+        .getRemoteSecondary()
+        ?.executeVerb(notifyFetchVerbBuilder);
+    if (atNotificationStr == null) {
```

Comment:
Need some more clarity on the potential exceptions here. Could have
* notification with that ID does not exist
* server-side exception of some sort in the cloud secondary
* failed to connect to cloud secondary

Also - if the notification does not exist, is that really an exception? Should we maybe just return null if the verb executes correctly but the notification is not found? (Just asking the question but I think my personal preference would be to receive a null for not found, and an exception if there is an exception.)

Reply:
In case of notification id does exist, servers return NotificationStatus.expired.

The above IF condition that checks for null is when remote secondary fails to respond back due to network issue or secondary not reacheable.